codegen/src/api/mod.rs

```diff
@@ -371,6 +375,42 @@ impl RuntimeGenerator {
             })
         });
 
+        let outer_error_variants = self.metadata.pallets.iter().filter_map(|p| {
+            let variant_name = format_ident!("{}", p.name);
+            let mod_name = format_ident!("{}", p.name.to_string().to_snake_case());
+            let index = proc_macro2::Literal::u8_unsuffixed(p.index);
+
+            p.error.as_ref().map(|_| {
+                quote! {
+                    #[codec(index = #index)]
+                    #variant_name(#mod_name::Error),
+                }
+            })
+        });
+
+        let outer_error = quote! {
+            #default_derives
+            pub enum Error {
+                #( #outer_error_variants )*
+            }
+        };
+
+        let root_error_match_arms = self.metadata.pallets.iter().filter_map(|p| {
+            let variant_index = p.index;
+            let variant_name_str = &p.name;
+            let variant_name = format_ident!("{}", variant_name_str);
+            let mod_name = format_ident!("{}", variant_name_str.to_string().to_snake_case());
+
+            p.error.as_ref().map(|_|
+                quote! {
+                    #variant_index => {
```

Comment:
Just as a nit; I did "if arms" and not "match arms" for the RootEvent stuff, so I think we should be consistent one way or the other :)

The reason I did "if arms" was so that the impl below that uses this output is cleaner (match arms mean you have part of the match syntax here and part of it below which imo is a little less clean).

Reply:
I'm not _strongly_ against the match style; the "if" one just felt nicer code wise; either way both impls should be identical to be consistent :)